### PR TITLE
Update built-ins to 1.88.1 level

### PR DESCRIPTION
**Pull request by tsmaeder**
#### What it does
Updates the built-in versions for `download:plugins` to 1.88.1 level. Also removes the explicit downloads of extensions which are included in the extension packs.

Contributed on behalf of ST Microelectronics

#### How to test
Delete the plugins folder in you Theia installation and run `yarn download:plugins`. The download should proceed normally and Theia should function as usual. 

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -102,11 +102,9 @@
   ],
   "theiaPluginsDir": "plugins",
   "theiaPlugins": {
-    "eclipse-theia.builtin-extension-pack": "https://open-vsx.org/api/eclipse-theia/builtin-extension-pack/1.83.1/file/eclipse-theia.builtin-extension-pack-1.83.1.vsix",
+    "eclipse-theia.builtin-extension-pack": "https://open-vsx.org/api/eclipse-theia/builtin-extension-pack/1.8.1/file/eclipse-theia.builtin-extension-pack-1.88.1.vsix",
```
Comment (jfaltermeier) [inline package.json:105]: I think the `/1.8.1/` should be `/1.88.1/`. Otherwise we get a 404

**Review by jfaltermeier — COMMENTED** [review]
One more thing I noticed is that when Theia is started with a workspace that does not have an open Editor, an error shown:

```
2024-05-02T08:45:17.912Z root ERROR [hosted-plugin: 185507] Activating extension 'TypeScript and JavaScript Language Features (built-in)' failed: Error: Assertion Failed: argument is undefined or null
    at assertIsDefined (/home/johannes/Git/theia/examples/electron/lib/backend/packages_plugin-ext_lib_common_plugin-api-rpc_js.js:2487:15)
    at get activeTabGroup [as activeTabGroup] (/home/johannes/Git/theia/examples/electron/lib/backend/node_modules_theia_monaco-editor-core_esm_vs_base_common_sync_recursive-packages_core_shared_-54ebfc.js:17515:72)
    at d.getEditorCandidatesForActiveTab (/home/johannes/Git/theia/plugins/vscode.typescript-language-features/extension/dist/extension.js:1:529555)
    at d.update (/home/johannes/Git/theia/plugins/vscode.typescript-language-features/extension/dist/extension.js:1:529301)
    at new d (/home/johannes/Git/theia/plugins/vscode.typescript-language-features/extension/dist/extension.js:1:529217)
    at t.activate (/home/johannes/Git/theia/plugins/vscode.typescript-language-features/extension/dist/extension.js:1:266847)
    at PluginManagerExtImpl.startPlugin (/home/johannes/Git/theia/examples/electron/lib/backend/node_modules_theia_monaco-editor-core_esm_vs_base_common_sync_recursive-packages_core_shared_-54ebfc.js:14867:81)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /home/johannes/Git/theia/examples/electron/lib/backend/node_modules_theia_monaco-editor-core_esm_vs_base_common_sync_recursive-packages_core_shared_-54ebfc.js:14774:21
    at async PluginManagerExtImpl.$activatePlugin (/home/johannes/Git/theia/examples/electron/lib/backend/node_modules_theia_monaco-editor-core_esm_vs_base_common_sync_recursive-packages_core_shared_-54ebfc.js:14828:13)
```

An error is logged as well, when the last editor is closed:
```
root ERROR [hosted-plugin: 182577] Error: Assertion Failed: argument is undefined or null
    at assertIsDefined (/home/johannes/Git/theia/examples/electron/lib/backend/packages_plugin-ext_lib_common_plugin-api-rpc_js.js:2487:15)
    at get activeTabGroup [as activeTabGroup] (/home/johannes/Git/theia/examples/electron/lib/backend/node_modules_theia_monaco-editor-core_esm_vs_base_common_sync_recursive-packages_core_shared_-54ebfc.js:17515:72)
    at d.getEditorCandidatesForActiveTab (/home/johannes/Git/theia/plugins/vscode.typescript-language-features/extension/dist/extension.js:1:529555)
    at d.update (/home/johannes/Git/theia/plugins/vscode.typescript-language-features/extension/dist/extension.js:1:529301)
    at /home/johannes/Git/theia/plugins/vscode.typescript-language-features/extension/dist/extension.js:1:529125
    at /home/johannes/Git/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:1410:69
    at CallbackList.invoke (/home/johannes/Git/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:1416:26)
    at Emitter.fire (/home/johannes/Git/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:1531:36)
    at /home/johannes/Git/theia/examples/electron/lib/backend/node_modules_theia_monaco-editor-core_esm_vs_base_common_sync_recursive-packages_core_shared_-54ebfc.js:19890:101
    at /home/johannes/Git/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:1410:69
```

**Comment by tsmaeder** [comment]
I filed an issue about the exception: it does not seem to have any effect, though.

**Comment by tsmaeder** [comment]
Seems the integration tests are not happy. I'll have a look.

**Review by jfaltermeier — APPROVED** [review]
LGTM, cleaning up the workaround is tracked on https://github.com/eclipse-theia/theia/issues/13679

**Pull request merged**